src/graphnet/models/data_representation/images/mappings/pixel_mappings.py

```diff
@@ -0,0 +1,398 @@
+
+        self._dtype = dtype
+        self._string_label = string_label
+        self._dom_number_label = dom_number_label
+        self._pixel_feature_names = pixel_feature_names
+
+        self._set_indeces(pixel_feature_names, dom_number_label, string_label)
+
+        self._nb_cnn_features = (
+            len(pixel_feature_names) - 2
+        )  # 2 for string and dom_number
+
+        self._include_main_array = include_main_array
+        self._include_lower_dc = include_lower_dc
+        self._include_upper_dc = include_upper_dc
+
+        # read mapping from parquet file
+        df = pd.read_parquet(IC86_CNN_MAPPING)
+        df.sort_values(
+            by=["string", "dom_number"],
+            ascending=[True, True],
+            inplace=True,
+        )
+
+        # Set the index to string and dom_number for faster lookup
+        df.set_index(
+            ["string", "dom_number"],
+            inplace=True,
+            drop=False,
+        )
+
+        self._mapping = df
+        super().__init__(pixel_feature_names=pixel_feature_names)
+
+    def _set_indeces(
+        self,
+        feature_names: List[str],
+        dom_number_label: str,
+        string_label: str,
+    ) -> None:
+        """Set the indices for the features."""
+        self._cnn_features_idx = []
+        for feature in feature_names:
+            if feature == dom_number_label:
+                self._dom_number_idx = feature_names.index(feature)
+            elif feature == string_label:
+                self._string_idx = feature_names.index(feature)
+            else:
+                self._cnn_features_idx.append(feature_names.index(feature))
+
+    def forward(self, data: Data, data_feature_names: List[str]) -> Data:
+        """Map pixel data to images."""
+        # Initialize output arrays
+        if self._include_main_array:
+            main_arr = torch.zeros(
+                (self._nb_cnn_features, 10, 10, 60),
+                dtype=self._dtype,
+            )
+        if self._include_upper_dc:
+            upper_dc_arr = torch.zeros(
+                (self._nb_cnn_features, 8, 10),
+                dtype=self._dtype,
+            )
+        if self._include_lower_dc:
+            lower_dc_arr = torch.zeros(
+                (self._nb_cnn_features, 8, 50),
+                dtype=self._dtype,
+            )
+
+        # data.x is expected to be a tensor with shape (N, F)
+        # where N is the number of nodes and F is the number of features.
+        x = data.x
+
+        # Direct coordinate and feature extraction
+        string_dom_number = x[
+            :, [self._string_idx, self._dom_number_idx]
+        ].int()
+        batch_row_features = x[:, self._cnn_features_idx]
+
+        # look up the mapping for string and dom_number
+        match_indices = self._mapping.loc[
+            zip(*string_dom_number.t().tolist())
+        ][
+            ["string", "dom_number", "mat_ax0", "mat_ax1", "mat_ax2"]
```

Comment:
Looks like your method relies on a very specific set of column names in this file.